alerts/cluster-monitoring-operator/TargetDown.md

```diff
@@ -0,0 +1,281 @@
+spec:
+  endpoints:
+  - bearerTokenSecret:
+      key: ""
+    interval: 15s
+    port: https
+    relabelings:
+    - action: replace
+      regex: (.*)
+      replacement: $1
+      sourceLabels:
+      - __meta_kubernetes_pod_node_name
+      targetLabel: instance
+    scheme: https
+    tlsConfig:
+      ca: {}
+      caFile: /etc/prometheus/configmaps/serving-certs-ca-bundle/service-ca.crt
+      cert: {}
+      certFile: /etc/prometheus/secrets/metrics-client-certs/tls.crt
+      keyFile: /etc/prometheus/secrets/metrics-client-certs/tls.key
+      serverName: node-exporter.openshift-monitoring.svc
+  jobLabel: app.kubernetes.io/name
+  namespaceSelector: {}
+  selector:
+    matchLabels:
+      app.kubernetes.io/component: exporter
+      app.kubernetes.io/name: node-exporter
+      app.kubernetes.io/part-of: openshift-monitoring
+```
+In the `Prometheus` that scrapes this target, the volume mounted at `/etc/prometheus/secrets/*`
+are the secrets having the same name as the subdirectory. In this example, the secret
+is `metrics-client-certs` in the namespace `openshift-monitoring`. Now we extract
+the certificate from the secret using this command:
+```bash
+oc extract secret/$SECRET_NAME -n $NAMESPACE --keys=tls.crt --to=- > certificate.crt
+```
+Then we inspect is expiration date.
+```bash
+openssl x509 -noout -enddate -in certificate.crt
+```
+The output should contain `notAfter` field as its expiration date.
+```bash
+notAfter=Aug  6 13:11:20 2025 GMT
+```
+
+Normally OpenShift **automatically renews** the certificates before expiration date.
+This date should be sometime in the future. If the certificate does expire without
+automatic renewal, please contact OpenShift support team.
+
+If the issue requires immediate resolution, please refer to [this guide on how to
+force a certificate renewal](https://access.redhat.com/solutions/5899121).
+
+To diagnose potential issues with automatic certificate renewal, perform the
+following checks:
+1. Ensure the `prometheus-k8s` stateful set logs do not display errors related to
+   certificates.
+   ```bash
+   oc logs statefulset/prometheus-k8s -n openshift-monitoring
+   ```
+2. Verify that the scraped target pod(s) logs are free from certificate-related errors.
+3. Check that the `cluster-monitoring-operator` pod is running and its logs contain
+   no errors regarding certificate rotation.
+   ```bash
+   oc logs deployment/cluster-monitoring-operator -n openshift-monitoring
+   ```
+4. Confirm the `prometheus-operator` pod is operational and its logs have no errors
+   concerning certificate rotation.
+   ```bash
+   oc logs deployment/prometheus-operator -n openshift-monitoring
+   ```
+
+
+#### Prometheus Scraping Interval
+
+Ensure that the scraping interval for the target is appropriately configured.
+Too frequent scraping can degrade the performance of target pod, even causing
+timeout when scraping.
+
+The last scrape duration is visible on the **Observe** -> **Targets** tab in the
```

Comment:
see picture
![scrape](https://github.com/juzhao/test/assets/22743651/b872068a-b982-4640-bc4e-a88b9469bf20)
it is the time of last scrape, not duration